tests/core/go_test/xmlreport_test.go

```diff
@@ -0,0 +1,130 @@
+// Copyright 2020 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package test_filter_test
+
+import (
+	"encoding/xml"
+	"io/ioutil"
+	"testing"
+
+	"github.com/bazelbuild/rules_go/go/tools/bazel_testing"
+)
+
+func TestMain(m *testing.M) {
+	bazel_testing.TestMain(m, bazel_testing.Args{
+		Main: `
+-- BUILD.bazel --
+load("@io_bazel_rules_go//go:def.bzl", "go_test")
+
+go_test(
+    name = "xml_test",
+    importpath = "github.com/bazelbuild/rules_go/tests/core/go_test/xml_test",
+    srcs = ["xml_test.go"],
+)
+
+-- xml_test.go --
+package test
+
+import (
+    "math/rand"
+    "testing"
+    "time"
+)
+
+func TestPass(t *testing.T) {
+    t.Parallel()
+}
+
+func TestPassLog(t *testing.T) {
+    t.Parallel()
+    t.Log("pass")
+}
+
+func TestFail(t *testing.T) {
+    t.Error("Not working")
+}
+
+func TestSubtests(t *testing.T) {
+    for i, subtest := range []string{"subtest a", "testB", "another subtest"} {
+        t.Run(subtest, func(t *testing.T) {
+            t.Logf("from subtest %s", subtest)
+            time.Sleep(time.Duration(rand.Intn(10)) * time.Millisecond)
+            t.Logf("from subtest %s", subtest)
+            if i%3 == 0 {
+                t.Skip("skipping this test")
+            }
+            if i%2 == 0 {
+                t.Fail()
+            }
+        })
+    }
+}
+`,
+	})
+}
+
+func Test(t *testing.T) {
+	err := bazel_testing.RunBazel("test", "//:xml_test")
+	if err == nil {
+		t.Fatal("expected bazel test to have failed")
+	}
+	if xerr, ok := err.(*bazel_testing.StderrExitError); !ok {
+		t.Fatalf("expected bazel tests to fail with exit code 3 (TESTS_FAILED), got: %s", err)
+	} else if xerr.Err.ExitCode() != 3 {
+		t.Fatalf("expected bazel tests to fail with exit code 3 (TESTS_FAILED), got: %s", xerr)
+	}
+	b, err := ioutil.ReadFile("bazel-out/darwin-fastbuild/testlogs/xml_test/test.xml")
```

Comment:
This test is failing on non-darwin platforms because the xml gets written somewhere else. `bazel info bazel-testlogs` should give you the right directory.

Reply:
Must have been half asleep while writing this :)